### PR TITLE
Validate that Elasticsearch home/data paths are readable

### PR DESCRIPTION
**Pull request by joschi**
These validations will ensure that cases like #2536 result in a better error message.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/configuration/ElasticsearchConfiguration.java [hunk 1]
```diff
@@ -268,4 +274,41 @@ public String getPathHome() {
...
+
+        if (Files.exists(parent) && !Files.isReadable(parent)) {
+            throw new ValidationException("Path " + parent.toAbsolutePath() + " cannot be read.");
+        }
+
+        if (Files.exists(path) && !Files.isDirectory(path)) {
+            throw new ValidationException("Path " + path.toAbsolutePath() + " is not a directory.");
+        }
+
+        if (Files.exists(path) && !Files.isReadable(path)) {
```
Comment (bernd) [inline graylog2-server/src/main/java/org/graylog2/configuration/ElasticsearchConfiguration.java]: Is there a case where the directory also needs to be writable? If so, we should also test for that.

Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/configuration/ElasticsearchConfiguration.java]: It has to be writable if data is being stored (`elasticsearch_node_data`). Other than that, I haven't found a case for which it had to be writable.

Funny enough, the directories don't even have to exist, it's sufficient if the parent directory is there.

Comment (bernd) [inline graylog2-server/src/main/java/org/graylog2/configuration/ElasticsearchConfiguration.java]: the :-1: was wrong :smile:

**Comment by bernd** [comment]
LGTM :+1:

File: graylog2-server/src/main/java/org/graylog2/configuration/ElasticsearchConfiguration.java (hunk 1 above)
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/configuration/ElasticsearchConfiguration.java]: [So la la](https://www.youtube.com/watch?v=imBvAPQCD4E)…

**Pull request merged**